core/security_group_test.go

```diff
@@ -0,0 +1,1128 @@
+				AllowAllOutbound: falsePtr,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_SecurityGroupCreateRequest_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *securityGroupCreateRequest
+	}{
+		{
+			name: "empty",
+			obj:  &securityGroupCreateRequest{},
+		},
+		{
+			name: "full",
+			obj: &securityGroupCreateRequest{
+				Organization: OrganizationRef{ID: "org_rs55YZNYMw7o3jnQ"},
+				Properties:   &SecurityGroupCreateArguments{Name: "group-1"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_SecurityGroupUpdateRequest_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *securityGroupUpdateRequest
+	}{
+		{
+			name: "empty",
+			obj:  &securityGroupUpdateRequest{},
+		},
+		{
+			name: "full",
+			obj: &securityGroupUpdateRequest{
+				SecurityGroup: SecurityGroupRef{ID: "sg_3uXbmANw4sQiF1J3"},
+				Properties:    &SecurityGroupUpdateArguments{Name: "updated"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_SecurityGroupsResponseBody_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *securityGroupsResponseBody
+	}{
+		{
+			name: "empty",
+			obj:  &securityGroupsResponseBody{},
+		},
+		{
+			name: "full",
+			obj: &securityGroupsResponseBody{
+				Pagination:     &katapult.Pagination{CurrentPage: 344},
+				SecurityGroup:  &SecurityGroup{ID: "id1"},
+				SecurityGroups: []*SecurityGroup{{ID: "id2"}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestSecurityGroupsClient_List(t *testing.T) {
+	// Correlates to fixtures/security_groups_list*.json
+	SecurityGroupList := []*SecurityGroup{
+		{
+			ID:           "sg_3uXbmANw4sQiF1J3",
+			Name:         "group-1",
+			Associations: []string{},
+		},
+		{
+			ID:           "sg_NFP2Ns2frZJV8gD1",
+			Name:         "group-2",
+			Associations: []string{},
+		},
+		{
+			ID:           "sg_FcIOv1SCf8366ZxJ",
+			Name:         "group-3",
+			Associations: []string{},
+		},
+	}
+
+	type args struct {
+		ctx  context.Context
+		org  OrganizationRef
+		opts *ListOptions
+	}
+	tests := []struct {
+		name           string
+		args           args
+		want           []*SecurityGroup
+		wantQuery      *url.Values
+		wantPagination *katapult.Pagination
+		errStr         string
+		errResp        *katapult.ResponseError
+		respStatus     int
+		respBody       []byte
+	}{
+		{
+			name: "by organization ID",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			want: SecurityGroupList,
+			wantQuery: &url.Values{
+				"organization[id]": []string{"org_O648YDMEYeLmqdmn"},
+			},
```

Comment:
`wantQuery` is not actually used within the test body, as it does this instead to validate the request query string:

```go
qs := queryValues(tt.args.org, tt.args.opts)
assert.Equal(t, *qs, r.URL.Query())
```

We currently have a bit of a mixture of `wantQuery` and the above method between different tests. I'm undecided over which method I prefer, though I think I'm leaning a little bit more towards the `wantQuery` variant as it makes the test case more obvious about what's going on, and relies less on the various `queryValues()` methods working correctly.

@strideynet, @andrewjtait, thoughts?

Though is probably a discussion that's outside of scope of this PR, and simply removing `wantQuery` from this test is fine to get this merged :)

Reply:
I think I lean towards the explicit testing of the query values (this is what I've done in tests using the FRM)